internal/controller/controllers/bmh_agent_controller.go

```diff
@@ -409,6 +452,17 @@ func (r *BMACReconciler) reconcileBMH(ctx context.Context, bmh *bmh_v1alpha1.Bar
 
 			bmh.ObjectMeta.Annotations[BMH_INSPECT_ANNOTATION] = "disabled"
 
+			if bmh.ObjectMeta.Annotations[BMH_DETACHED_ANNOTATION] == "true" {
```

Comment:
According to the docs, [delete](https://golang.org/pkg/builtin/#delete) will be a no-op if the key doesn't exist or if the map is nil. I think we can remove this `if` statement.

Not having this `if` will protect us from user-errors where the value is set to (for whatever reason) something else that is not `true`. BMO doesn't check the value, so we probably shouldn't do that either:

https://github.com/metal3-io/baremetal-operator/blob/a39c5d3866ee3554417e11075f535b13275ff396/controllers/metal3.io/host_state_machine.go#L242-L250

Reply:
I will remove the if statement.